base/linalg/diagonal.jl

```diff
@@ -135,16 +135,16 @@ function tril!(D::Diagonal,k::Integer=0)
     return D
 end
 
-==(Da::Diagonal, Db::Diagonal) = Da.diag == Db.diag
--(A::Diagonal)=Diagonal(-A.diag)
-+(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag + Db.diag)
--(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag - Db.diag)
+(==)(Da::Diagonal, Db::Diagonal) = Da.diag == Db.diag
```

Comment:
`+(x::Int, y::Int)` has got to be one of the ugliest hacks, because it initially parses as `(+)((x::Int, y::Int))` and then there's special case code "correcting" this to `(+)(x::Int, y::Int)`. It breaks down upon the smallest pertubation; e.g. defining `+(x::Int, y::Int)::Int = 1` just doesn't work. Avoiding this syntax whenever possible seems like a good idea.